src/mat.c

```diff
@@ -52,31 +52,36 @@
 #   include "mat73.h"
 #endif
 
+#if defined(_WIN64) || defined(_WIN32)
+#include <fcntl.h>
+#include <share.h>
+#endif
+
 /*
  *===================================================================
  *                 Private Functions
  *===================================================================
  */
 
-#if defined(_WIN64) || defined(_WIN32)
-#include <fcntl.h>
-#include <share.h>
-
-static int
-mkstemp( char *templ )
+static char *
+TempFilename()
 {
-  int maxtry = 26, rtn = -1;
-
-  while( maxtry-- && (rtn < 0) )
-  {
-    char *r = _mktemp( templ );
-    if( r == NULL )
-      return -1;
-    rtn = sopen( r, O_RDWR | O_CREAT | O_EXCL | O_BINARY, SH_DENYRW, 0600 );
+  static char template[7] = "XXXXXX";
+  static char temp[7];
+  strcpy(temp, template);
+#if defined(_WIN64) || defined(_WIN32)
+  return _mktemp(temp);
+#else
+  int fd = mkstemp(temp);
+  if (fd > 0) {
+    close(fd);
+    return temp;
```

Comment:
Not sure that the fwrite vs write is the perfect fix either since I don't think it will work with the hdf5-based file formats will it?